src/D2L.Bmx/UnixParentProcess.cs

```diff
@@ -0,0 +1,47 @@
+using System.Diagnostics;
+
+namespace D2L.Bmx;
+
+internal class UnixParentProcess {
+
+	public static string GetParentProcessName() {
+		int parentPid = GetParentProcessId( GetParentProcessId( Process.GetCurrentProcess().Id ) );
+		var proccessStartInfo = new ProcessStartInfo {
+			FileName = "/bin/ps",
+			ArgumentList = { "-p", $"{parentPid}", "-o", "comm=" },
```

Comment:
given the id, you can get the name (and other info) of a process by using standard .NET APIs
see https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.process.getprocessbyid